plugin/system_copy.vim

```diff
@@ -18,7 +18,14 @@ function! s:system_copy(type, ...) abort
   else
     silent exe "normal! `[v`]y"
   endif
-  silent call system('pbcopy', getreg('@'))
+  let os = substitute(system('uname'), '\n', '', '')
+  if has("gui_mac") || os == 'Darwin'
+    silent call system('pbcopy', getreg('@'))
+  elseif has("gui_win32")
+    silent call system('clip', getreg('@')) 
+  else
+    silent call system('xsel --clipboard --input', getreg('@')) 
+  endif
```

Comment:
Can you extract this such that there is a function which returns the OS dependant copy command, and use that in here?

``` vim
   silent call system(s:CopyCommandForCurrentOS(), getreg('@'))
endif

function! s:CopyCommandForCurrentOS()
  let os = substitute(system('uname'), '\n', '', '')
  if has("gui_mac") || os == 'Darwin'
    return 'pbcopy'
  elseif has("gui_win32")
    return 'clip'
  else
    return 'xsel --clipboard --input'
  endif
endfunction
```